lib/rubocop/cop/rails/active_model_errors_direct_manipulation.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks direct manipulation of ActiveModel#errors as hash.
+      # These operations are deprecated in Rails 6.1 and will not work in Rails 7.
+      #
+      # @example
+      #   # bad
+      #   user.errors[:name] << 'msg'
+      #   user.errors.messages[:name] << 'msg'
+      #
+      #   # good
+      #   user.errors.add(:name, 'msg')
+      #
+      #   # bad
+      #   user.errors[:name].clear
+      #   user.errors.messages[:name].clear
+      #
+      #   # good
+      #   user.errors.delete(:name)
+      #
+      class ActiveModelErrorsDirectManipulation < Base
+        MSG = 'Avoid manipulating ActiveModel errors as hash directly.'
+
+        MANIPULATIVE_METHODS = ':<< :append :clear :collect! :compact! :concat '\
+        ':delete :delete_at :delete_if :drop :drop_while :fill :filter! :keep_if '\
+        ':flatten! :insert :map! :pop :prepend :push :reject! :replace :reverse! '\
+        ':rotate! :select! :shift :shuffle! :slice! :sort! :sort_by! :uniq! :unshift'
+
+        def_node_matcher :active_model_errors_root_manipulation?, <<~PATTERN
+          (send
+            (send
+              (send {send ivar lvar} :errors)
+              :[]
+              ...)
+            {#{MANIPULATIVE_METHODS}}
+            ...)
+        PATTERN
+
+        def_node_matcher :active_model_errors_root_assignment?, <<~PATTERN
+          (send
+            (send {send ivar lvar} :errors)
+            :[]=
+            ...)
+        PATTERN
+
+        def_node_matcher :active_model_errors_manipulation?, <<~PATTERN
+          (send
+            (send
+              (send
+                (send {send ivar lvar} :errors)
+                {:messages :details})
+              :[]
+              ...)
+            {#{MANIPULATIVE_METHODS}}
+            ...)
+        PATTERN
+
+        def_node_matcher :active_model_errors_assignment?, <<~PATTERN
+          (send
+            (send
+              (send {send ivar lvar} :errors)
+              {:messages :details})
+            :[]=
+            ...)
+        PATTERN
+
+        def on_send(node)
+          add_offense(node) if active_model_errors_root_assignment?(node)
+          add_offense(node) if active_model_errors_root_manipulation?(node)
+          add_offense(node) if active_model_errors_manipulation?(node)
+          add_offense(node) if active_model_errors_assignment?(node)
```

Comment:
Maybe this `add_offense` is enough only once?

```suggestion
          if active_model_errors_root_assignment?(node) ||
             active_model_errors_root_manipulation?(node) ||
             active_model_errors_manipulation?(node) ||
             active_model_errors_assignment?(node)
            add_offense(node) 
          end            
```